src/Features/CSharp/Portable/UpgradeProject/CSharpUpgradeProjectCodeFixProvider.cs

```diff
@@ -41,6 +41,16 @@ public override string SuggestedVersion(ImmutableArray<Diagnostic> diagnostics)
             return RequiredVersion(diagnostics).ToDisplayString();
         }
 
+        public override string AddBetaIfNeeded(string version)
```

Comment:
Should this just be a compiler API?

Reply:
Since we'll remove this in whatever 16.x where C# 8 RTMs, I would rather keep this as presentation-only (if we add a public API, then we'd have to keep it).

Reply:
From our discussion, a possible API that the compiler could expose is list language versions and how to display them (which is an existing problem which this PR doesn't try to solve). That would avoid fiddling with different layers, and that is tracked in https://github.com/dotnet/project-system/issues/3121